### PR TITLE
Added Dark Mode as a component

### PR DESCRIPTION
**Pull request by ArunPurewal1994**
Added dark mode/light mode button

### CODE REVIEW

File: src/lib/DarkMode.svelte [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			(!localStorage.theme && window.matchMedia('(prefers-color-scheme: dark)').matches)
+		) {
+			document.documentElement.classList.add('dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+		}
+	</script>
+</svelte:head>
+
+<button
```
Comment (jamesbirtles) [inline src/lib/DarkMode.svelte:48]: We should create an issue to create a Switch component, then use that here. Theres a few accessibility requirements to consider for switches that this currently doesn't handle - https://www.w3.org/WAI/ARIA/apg/patterns/switch/

Unfortunately the switch in headless ui (https://captaincodeman.github.io/svelte-headlessui/switch/) is not implemented yet - might be a good contribution to make

File: src/lib/accordion/Accordion.svelte [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 <div
 	class={classes(
-		'border-gray-200 bg-white',
```
Comment (jamesbirtles) [inline src/lib/accordion/Accordion.svelte:22]: i'd rather keep the background on this, a transparent accordion would be weird imo

File: src/routes/DarkMode/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<script lang="ts">
+	import DarkMode from '$lib/DarkMode.svelte';
+	import ExamplePage from '../ExamplePage.svelte';
+
+	const usage = '<DarkMode />';
```
Comment (jamesbirtles) [inline src/routes/DarkMode/+page.svelte:5]: Even though its a pretty basic usage, I still think its worth having the extra component and importing usage using `?raw`
e.g.
```
import ExampleDarkMode from "./ExampleDarkMode.svelte";
import usage from "./ExampleDarkMode.svelte?raw";
```

**Review by jamesbirtles — CHANGES_REQUESTED** [review]

File: src/app.css [hunk 4]
```diff
@@ -1,3 +1,8 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+html,
+body {
+	@apply bg-white dark:bg-gray-900 dark:text-white;
```
[the remark this answers is not in the document]
Comment (jamesbirtles) [reply src/app.css:7]: it's worth reading this page - https://tailwindcss.com/docs/reusing-styles
essentially `@apply` should be avoided

File: src/lib/DarkMode.svelte [hunk 5]
```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+
+	let dark: boolean;
```
[the remark this answers is not in the document]
Comment (ArunPurewal1994) [reply src/lib/DarkMode.svelte:4]: I've specified it as boolean as this part of the code here: 
dark = document.documentElement.classList.contains('dark');  cannot be determined as false as it's initial state

[the remark this answers is not in the document]
Comment (jamesbirtles) [reply src/lib/DarkMode.svelte:4]: the problem is that on the server the onMount will never be run, and so I think the variable will technically be undefined if we don't initialise it to something

File: src/app.d.ts [hunk 6]
```diff
@@ -1,6 +1,4 @@
 declare namespace App {
-	//interface Locals {}
-
 	interface Platform {
 		//env: {};
 		context: {
```
Comment (danny-people-plus) [inline src/app.d.ts:4]: ```<never>```
Instead of
```<any>```
?

File: src/app.html [hunk 7]
```diff
@@ -6,7 +6,7 @@
 		<meta name="viewport" content="width=device-width, initial-scale=1" />
 		%sveltekit.head%
 	</head>
-	<body class="h-full">
+	<body class="h-full bg-white dark:bg-gray-900 text-gray-900 dark:text-white">
```
Comment (jamesbirtles) [inline src/app.html:9]: the background is set on the html tag above to `bg-gray-100`. The duplication should be removed, and i'd prefer the background color remain the same as it was before

File: src/lib/DarkMode.svelte [hunk 8]
```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+
+	let dark = false;
+	let hidden = true;
```
Comment (jamesbirtles) [inline src/lib/DarkMode.svelte:5]: whats the reason for this component being hidden until mounted?

File: src/routes/DarkMode/ExampleDarkMode.svelte [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<script lang="ts">
+	import DarkMode from '$lib/DarkMode.svelte';
```
Comment (jamesbirtles) [inline src/routes/DarkMode/ExampleDarkMode.svelte:2]:
```suggestion
	import { DarkMode } from '$lib';
```

File: src/routes/+layout.svelte [hunk 10]
```diff
@@ -45,7 +49,7 @@
 		</nav>
 
 		<div class="grow overflow-x-hidden">
-			<div class="h-full p-6 bg-white border border-gray-200 rounded-md overflow-y-auto">
+			<div class="h-full p-6 border border-gray-200 rounded-md overflow-y-auto">
```
Comment (jamesbirtles) [inline src/routes/+layout.svelte:52]: i don't think bg-white should be removed

**Review by jamesbirtles — CHANGES_REQUESTED** [review]
A few more comments. I also clicked the button to merge uat into this PR, so you'll want to `git pull` before making any further changes

**Review by jamesbirtles — APPROVED** [review]

**Pull request merged**